app/vmctl/opentsdb/opentsdb.go

```diff
@@ -0,0 +1,352 @@
+package opentsdb
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// Retention objects contain meta data about what to query for our run
+type Retention struct {
+	/*
+		OpenTSDB has two levels of aggregation,
+		First, we aggregate any un-mentioned tags into the last result
+		Second, we aggregate into buckets over time
+		To simulate this with config, we have
+		FirstOrder (e.g. sum/avg/max/etc.)
+		SecondOrder (e.g. sum/avg/max/etc.)
+		AggTime	(e.g. 1m/10m/1d/etc.)
+		This will build into m=<FirstOrder>:<AggTime>-<SecondOrder>-none:
+		Or an example: m=sum:1m-avg-none
+	*/
+	FirstOrder  string
+	SecondOrder string
+	AggTime     string
+	// The actual ranges will will attempt to query (as offsets from now)
+	QueryRanges []TimeRange
+}
+
+// RetentionMeta objects exist to pass smaller subsets (only one retention range) of a full Retention object around
+type RetentionMeta struct {
+	FirstOrder  string
+	SecondOrder string
+	AggTime     string
+}
+
+// Client object holds general config about how queries should be performed
+type Client struct {
+	Addr string
+	// The meta query limit for series returned
+	Limit      int
+	Retentions []Retention
+	Filters    []string
+	Normalize  bool
+}
+
+// Config contains fields required
+// for Client configuration
+type Config struct {
+	Addr       string
+	Limit      int
+	Offset     int64
+	HardTS     int64
+	Retentions []string
+	Filters    []string
+	Normalize  bool
+	MsecsTime  bool
+}
+
+// TimeRange contains data about time ranges to query
+type TimeRange struct {
+	Start int64
+	End   int64
+}
+
+// MetaResults contains return data from search series lookup queries
+type MetaResults struct {
+	Type    string `json:"type"`
```

Comment:
Or this field `Type` which seems unused.

Reply:
But I may be missing something or it will be useful to leave it as docs for opentsdb responses.